pkg/download/download.go

```diff
@@ -82,6 +83,12 @@ func Download(endpoint string, opts ...Option) (b []byte, err error) {
 	}
 
 	err = retry.Exponential(60*time.Second, retry.WithUnits(time.Second), retry.WithJitter(time.Second)).Retry(func() error {
+		select {
+		case <-ctx.Done():
+			return context.Canceled
```

Comment:
it might make sense to add here `retry.UnexpectedError(context.Canceled)` or `retry.UnexpectedError(context.Error())`